nova/tests/unit/scheduler/weights/test_weights_resize_same_host.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2020 OpenStack Foundation
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+Tests for scheduler prefer-resize-to-same-host weigher.
+"""
+from nova import objects
+from nova.scheduler import weights
+from nova.scheduler.weights import resize_same_host
+from nova import test
+from nova.tests.unit.scheduler import fakes
+
+
+class PreferSameHostOnResizeWeigherTestCase(test.NoDBTestCase):
+    def setUp(self):
+        super(PreferSameHostOnResizeWeigherTestCase, self).setUp()
+        self.weight_handler = weights.HostWeightHandler()
+        self.weighers = [resize_same_host.PreferSameHostOnResizeWeigher()]
+
+    def _prepare_instances_and_hosts(self):
+        flavor_big = objects.Flavor(name='big', memory_mb=1024 * 512,
+                                    extra_specs={})
+        flavor_small = objects.Flavor(name='small', memory_mb=1024 * 256,
+                                      extra_specs={})
+        instances = [
+            objects.Instance(uuid='00000000-0000-4000-0000-000000000001',
```

Comment:
You shoud use `uuidsentinel` instead of hardcoding uuids. It keeps the same uuid for the same name. `grep` a little to see usage of it.